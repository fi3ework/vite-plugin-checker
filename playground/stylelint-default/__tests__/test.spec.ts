import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'
import {
  diagnostics,
  editFile,
  expectStderrContains,
  isBuild,
  isServe,
  log,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
} from '../../testUtils'

describe('stylelint', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      await sleepForServerReady()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit error file --')
      resetReceivedLog()
      editFile('src/style.css', (code) => code.replace(`color: rgb(0, 0, 0);`, `color: #fff;`))
      await sleepForEdit()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    const expectedMsg = ['Unexpected empty block', 'Expected modern color-function notation']

    it('should fail', async () => {
      expectStderrContains(log, expectedMsg)
    })
  })
})
