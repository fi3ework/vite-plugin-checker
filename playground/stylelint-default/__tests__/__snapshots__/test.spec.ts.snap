// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stylelint > serve > get initial error and subsequent error 1`] = `"[{"checkerId":"Stylelint","frame":"  > 1 | #root {/n      |       ^/n  > 2 | }/n      | ^^/n    3 |/n    4 | body {/n    5 |   color: rgb(0, 0, 0);","id":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","level":1,"loc":{"column":7,"file":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","line":1},"message":"Unexpected empty block (block-no-empty) (block-no-empty)","stack":""},{"checkerId":"Stylelint","frame":"    3 |/n    4 | body {/n  > 5 |   color: rgb(0, 0, 0);/n      |          ^^^^^^^^^^^^/n    6 | }/n    7 |","id":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","level":1,"loc":{"column":10,"file":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","line":5},"message":"Expected modern color-function notation (color-function-notation) (color-function-notation)","stack":""}]"`;

exports[`stylelint > serve > get initial error and subsequent error 2`] = `
[
  " ERROR(Stylelint)  Unexpected empty block (block-no-empty) (block-no-empty)
 FILE  <PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css:1:7

  > 1 | #root {
      |       ^
  > 2 | }
      | ^^
    3 |
    4 | body {
    5 |   color: rgb(0, 0, 0);
",
  " ERROR(Stylelint)  Expected modern color-function notation (color-function-notation) (color-function-notation)
 FILE  <PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css:5:10

    3 |
    4 | body {
  > 5 |   color: rgb(0, 0, 0);
      |          ^^^^^^^^^^^^
    6 | }
    7 |
",
  "[Stylelint] Found 2 errors and 0 warning",
]
`;

exports[`stylelint > serve > get initial error and subsequent error 3`] = `"[{"checkerId":"Stylelint","frame":"  > 1 | #root {/n      |       ^/n  > 2 | }/n      | ^^/n    3 |/n    4 | body {/n    5 |   color: rgb(0, 0, 0);","id":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","level":1,"loc":{"column":7,"file":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","line":1},"message":"Unexpected empty block (block-no-empty) (block-no-empty)","stack":""},{"checkerId":"Stylelint","frame":"    3 |/n    4 | body {/n  > 5 |   color: rgb(0, 0, 0);/n      |          ^^^^^^^^^^^^/n    6 | }/n    7 |","id":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","level":1,"loc":{"column":10,"file":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","line":5},"message":"Expected modern color-function notation (color-function-notation) (color-function-notation)","stack":""},{"checkerId":"Stylelint","frame":"  > 1 | #root {/n      |       ^/n  > 2 | }/n      | ^^/n    3 |/n    4 | body {/n    5 |   color: #fff;","id":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","level":1,"loc":{"column":7,"file":"<PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css","line":1},"message":"Unexpected empty block (block-no-empty) (block-no-empty)","stack":""}]"`;

exports[`stylelint > serve > get initial error and subsequent error 4`] = `
[
  " ERROR(Stylelint)  Unexpected empty block (block-no-empty) (block-no-empty)
 FILE  <PROJECT_ROOT>/playground-temp/stylelint-default/src/style.css:1:7

  > 1 | #root {
      |       ^
  > 2 | }
      | ^^
    3 |
    4 | body {
    5 |   color: #fff;
",
  "[Stylelint] Found 1 error and 0 warning",
]
`;
